lib/utils/url-utils.js

```diff
@@ -19,4 +21,15 @@ function init() {
   state = { ...state, isRunningElectron, openExternalUrl };
 }
 
-export const viewExternalUrl = url => state.openExternalUrl(url);
+export const viewExternalUrl = url => {
+  const lowerCaseUrl = url.toLowerCase().trim();
+  const urlStartsWithProtocol = function(protocol) {
+    return this.startsWith(protocol);
+  };
+
+  if (!allowedProtocols.some(urlStartsWithProtocol, lowerCaseUrl)) {
+    return;
+  }
```

Comment:
we may want to use something like `URL` instead which does actual URL parsing…

```js
const protocol = new URL( url ).protocol;
if ( ! allowedProtocols.some( allowed => allowed === protocol ) ) {
	return;
}
```

Reply:
Nice! Updated in b3e2d2bfbac66fdbc93f3256d213d333db8aa923. Unfortunately it doesn't look like you can use fat arrow functions to support `this`.

Reply:
you're right, but we don't even need `this`…

```js
if ( ! allowedProtocols.some( allowed => allowed === protocol ) ) {
	return;
}
```

this is because `Array.prototyp.some()` takes a callback and it passes in as the first argument of that callback the current value of the array being iterated over.

```js
allowedProtocols.some( ( currentValue, indexOfArray, originalArray ) => someTruthyOrFalsyReturnValue() )
```

Reply:
I see! Could have sworn I tried that... anyways it works great.